querydsl-sql/src/main/java/com/querydsl/sql/AbstractSQLQuery.java

```diff
@@ -251,10 +251,23 @@ protected void endContext(SQLListenerContext context) {
      *
      * @param exprs the expression arguments to retrieve
      * @return results as ResultSet
+     * @deprecated Use @{code select(..)} to define the projection and {@code getResults()} to obtain
+     *             the result set
      */
+    @Deprecated
     public ResultSet getResults(Expression<?>... exprs) {
-        queryMixin.setProjection(exprs);
+        if (exprs.length > 0) {
+            queryMixin.setProjection(exprs);
+        }
+        return getResults();
+    }
 
+    /**
+     * Get the results as an JDBC result set
```

Comment:
I would turn this into
`Get the results as a JDBC result set`
or
`Get the results as a JDBC ResultSet`